lochness/box/__init__.py

```diff
@@ -0,0 +1,365 @@
+import os, sys
+import gzip
+import logging
+import importlib
+import boxsdk
+import lochness
+import tempfile as tf
+import cryptease as crypt
+import lochness.net as net
+from typing import Generator, Tuple
+from pathlib import Path
+import hashlib
+from io import BytesIO
+import lochness.keyring as keyring
+import lochness.net as net
+import lochness.tree as tree
+from os.path import join, basename
+from boxsdk import Client, OAuth2
+import cryptease as enc
+import re
+
+
+logger = logging.getLogger(__name__)
+Module = lochness.lchop(__name__, 'lochness.')
+Basename = lochness.lchop(__name__, 'lochness.box.')
+
+CHUNK_SIZE = 65536
+
+def delete_on_success(Lochness, module_name):
+    ''' get module-specific delete_on_success flag with a safe default '''
+    value = Lochness.get('box', dict()) \
+                    .get(module_name, dict()) \
+                    .get('delete_on_success', False)
+    # if this is anything but a boolean, just return False
+    if not isinstance(value, bool):
+        return False
+    return value
+
+
+def base(Lochness, module_name):
+    ''' get module-specific base box diretory '''
+    return Lochness.get('box', {}) \
+                   .get(module_name, {}) \
+                   .get('base', '')
+
+
+def get_box_object_based_on_name(client:boxsdk.client,
+                                 box_folder_name: str,
+                                 box_path_id: str = '0') \
+                                         -> boxsdk.object.folder:
+    '''Return Box folder object for the given folder name
+
+    Currently, there is no api function to get box folder objects using
+    path strings in Box.
+        - https://stackoverflow.com/questions/16153409/is-there-any-easy-way-to-get-folderid-based-on-a-given-path
+
+    This function will recursively search for a folder which
+    has the same name as the `bx_head` and return its id.
+
+    Key Arguments:
+        client: Box client object
+        box_folder_name: Name of the folder in interest, str
+        box_path: Known parent id, str of int.
+                  Default=0. This execute search from the root.
+
+    Returns:
+        box_folder_object
+    '''
+    box_folder_name = str(box_folder_name)
+
+    # get list of files and directories under the top directory
+    root_dir = client.folder(folder_id=box_path_id).get()
+
+    # for entry in listing.entries:
+    for file_or_folder in root_dir.get_items():
+        if file_or_folder.type == 'folder' and \
+           file_or_folder.name == box_folder_name:
+            return file_or_folder
+
+
+def walk_from_folder_object(root: str, box_folder_object) -> \
+        Generator[str, list, list]:
+    '''top-down os.path.walk that operates on a Box folder object
+
+    Box does not support getting files with path strings.
```

Comment:
Should be worth mentioning what it supports.

Reply:
ef40e8d348762097b6192912aa5d6a2152f98493
Here it is!